### PR TITLE
Adding syslog default rules

### PR DESCRIPTION
**Pull request by d200201123**
Syslog rules to be shipped as part of Healthbot

### CODE REVIEW

**Comment by prashant3167** [comment]
jenkins please retry a build

File: juniper_official/Solutions/Pfe-wedge-detection/chassisd-pattern-set.rule [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    ingest-settings {
+        syslog {
+            pattern CHASSISD_EVENT1 {
+                event-id CHASSISD_EVENT1;
+                filter "fpc%{NUMBER:fpc_number}.*CHASSISD_I2CS_READBACK_ERROR.*Readback error from I2C slave for FPC.*";
+                description "This is transient error caused by parity error detected in the I2C response while reading a device";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
```
Comment (ConstrainedRandomVar) [inline juniper_official/Solutions/Pfe-wedge-detection/chassisd-pattern-set.rule]: Why are we keeping this count only a message? Can we not write this as a rule using the count and then display this in the trigger when the count goes beyond 4?

File: juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule [hunk 2]
```diff
@@ -0,0 +1,489 @@
...
+            }
+            pattern EACHIP_EVENT003 {
+                event-id EACHIP_EVENT003;
+                filter "fpc%{NUMBER:fpc_number}.*MQSS.*DRD.*Protect.*Parity error corrected for alloc state memory ";
+                description "The \"MQSS alloc state memory\" message reports a parity error correction done in the EACHIP state memory. This is due to a software micro-code error or transient hardware error";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
+                    value "If single log, clear alarm from CLI ('clear chassis fpc errors fpc-slot <fpc> error-code <code>', where <code> is from 'show chassis alarms').  If multiple logs, reset FPC and monitor.";
```
Comment (ConstrainedRandomVar) [inline juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule]: In addition to just flagging the description, can we implement this in the rule as a count to suggest resetting FPC as part of the trigger message?

File: juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule [hunk 3]
```diff
@@ -0,0 +1,489 @@
...
+            }
+            pattern EACHIP_EVENT004 {
+                event-id EACHIP_EVENT004;
+                filter "fpc%{NUMBER:fpc_number}.*MQSS.*LI.*Unroll TAIL length overflow";
+                description "The \"Unroll TAIL length overflow\" message reports that the packet size used in the forwarding path at a certain stage is greater than the original packet size. This is likely due to a software micro-code error.";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
+                    value "Reset FPC immediately and monitor for recurrence.  If it recurs, then RMA.";
```
Comment (ConstrainedRandomVar) [inline juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule]: Can we use a count based rule here as well?

File: juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule [hunk 4]
```diff
@@ -0,0 +1,489 @@
...
+                }
+                constant syslog-trigger-message {
+                    value "If single log, no action.  If multiple errors at same location after minor alarm is raised, contact JTAC.";
+                }                       
+                key-fields fpc_number;
+            }
+            pattern EACHIP_EVENT017 {
+                event-id EACHIP_EVENT017;
+                filter "fpc%{NUMBER:fpc_number}.*MQSS.*DRD.*CMD FSM state error";
+                description "The \"CMD FSM state error\" message reports that the Forwarding State Machine of the reorder queue received an unexpected value. This indicates the lookup microcode within LUSS is sending an incorrect command parcel to the MQSS Center Chip.";
```
Comment (ConstrainedRandomVar) [inline juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule]: How do such long descriptions show up on the GUI? Are they readable?

File: juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule [hunk 5]
```diff
@@ -0,0 +1,489 @@
...
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
```
Comment (ConstrainedRandomVar) [inline juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule]: Please add a count function to track the number of occurrences.

File: juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule [hunk 6]
```diff
@@ -0,0 +1,489 @@
...
+                key-fields fpc_number;
+            }
+            pattern EACHIP_EVENT024 {
+                event-id EACHIP_EVENT024;
+                filter "fpc%{NUMBER:fpc_number}.*EA.*HMCIF Rx: Link.*A response packet with a FATAL state is received from HMC";
+                description "The \"response packet with a FATAL state is received from HMC with State 0x1f\" message reports a multi-bit uncorrectable error condition in the Hybrid Memory Cube. This indicates that the internal logic in the Hybrid Memory Cube has encountered a multi-bit uncorrectable error condition from which it cannot recover without a reset.";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
```
Comment (ConstrainedRandomVar) [inline juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule]: Can we add a count function to track reccurrence?

File: juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule [hunk 7]
```diff
@@ -0,0 +1,489 @@
...
+                key-fields fpc_number;
+            }
+            pattern EACHIP_EVENT036 {
+                event-id EACHIP_EVENT036;
+                filter "fpc%{NUMBER:fpc_number}.*MQSS.*FI.*Parity error detected for bank.*of CP table";
+                description "The \"Parity error detected for bank .. of CP table\" message reports a hardware parity error detected in the EACHIP Fabric input block. This is due to a transient hardware SRAM error within EACHIP Fabric input block.";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
```
Comment (ConstrainedRandomVar) [inline juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule]: Should we also have a count based rule for this?

File: juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule [hunk 8]
```diff
@@ -0,0 +1,489 @@
...
+            }
+            pattern EACHIP_EVENT037 {   
+                event-id EACHIP_EVENT037;
+                filter "fpc%{NUMBER:fpc_number}.*HMM.*Lane.*TX RLL event ";
+                description "The \"TX RLL event\" message reports Run Length Limit (RLL) event on a Hybrid Memory Link. This indicates a Run Length Limit event on scrambled links and Junos OS will automatically retry to recover. Successful recovery will be confirmed with the second debug message with State: values between 0x20 - 0x23.​ A response packet with a benign WARNING state is received from HMC - State: 0x20";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
+                    value "No action if only a few occurrences.  If seen repeatedly, contact JTAC.";
```
Comment (ConstrainedRandomVar) [inline juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule]: And here..

File: juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule [hunk 9]
```diff
@@ -0,0 +1,489 @@
...
+            }
+            pattern EACHIP_EVENT038 {
+                event-id EACHIP_EVENT038;
+                filter "fpc%{NUMBER:fpc_number}.*MQSS.*BCMF CBUF.*Parity error corrected for buffer free list memory for bank";
+                description "The \"Parity error corrected for buffer free list memory for bank\" message reports a transient hardware error which was automatically corrected. This is due to a transient hardware SRAM error with EACHIP.";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
+                    value "If single log, clear alarm from CLI ('clear chassis fpc errors fpc-slot <fpc> error-code <code>', where <code> is from 'show chassis alarms').  If multiple logs, reset FPC and monitor.";
```
Comment (ConstrainedRandomVar) [inline juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule]: And here..

File: juniper_official/Solutions/Pfe-wedge-detection/lueachip-pattern-set.rule [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+    ingest-settings {
+        syslog {
+            pattern LUEA_EVENT001 {
+                event-id LUEA_EVENT001;
+                filter "fpc%{NUMBER:fpc_number}.*lmem addr error";
+                description "The \"lmem addr error\" message is caused by a transient hardware or software defect. This error is caused by a transient hardware or software defect. ";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
```
Comment (ConstrainedRandomVar) [inline juniper_official/Solutions/Pfe-wedge-detection/lueachip-pattern-set.rule]: And here..

File: juniper_official/Solutions/Pfe-wedge-detection/lueachip-pattern-set.rule [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+                key-fields fpc_number;
+            }
+            pattern LUEA_EVENT005 {
+                event-id LUEA_EVENT005;
+                filter "fpc%{NUMBER:fpc_number}.*Uninitialized Read Error";
+                description "The \"Uninitialized Read Error\" message is caused by a transient hardware or software error.Access to unintialized memory with predefined patterns are software errors and the rest are hardware errors.";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
```
Comment (ConstrainedRandomVar) [inline juniper_official/Solutions/Pfe-wedge-detection/lueachip-pattern-set.rule]: And here..

File: juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set.rule [hunk 12]
```diff
@@ -0,0 +1,201 @@
...
+    ingest-settings {
+        syslog {
+            pattern MQCHIP_EVENT004 {
+                event-id MQCHIP_EVENT004;
+                filter "fpc%{NUMBER:fpc_number}.*MQCHIP.*PT CPT parity error detected";
+                description "The \"PT CPT parity error detected\" message reports a transient hardware issue. The cause is due to a transient hardware issue occurring on the PFE, which affects CPT memory.";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
```
Comment (ConstrainedRandomVar) [inline juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set.rule]: Add count here.

File: juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set.rule [hunk 13]
```diff
@@ -0,0 +1,201 @@
...
+                key-fields fpc_number;
+            }
+            pattern MQCHIP_EVENT007 {
+                event-id MQCHIP_EVENT007;
+                filter "fpc%{NUMBER:fpc_number}.*MQCHIP.*WI parity error detected when read from ibuf";
+                description "The \"WI parity error detected\" message reports a transient hardware issue. The cause is due to a parity error detected in the internal MQCHIP memory.";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
```
Comment (ConstrainedRandomVar) [inline juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set.rule]: Add count based rule here.

File: juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set.rule [hunk 14]
```diff
@@ -0,0 +1,201 @@
...
+                key-fields fpc_number;
+            }
+            pattern MQCHIP_EVENT008 {
+                event-id MQCHIP_EVENT008;
+                filter "fpc%{NUMBER:fpc_number}.*MQCHIP.*CPQ RLDRAM single bit ECC error.*bank.*addr";
+                description "The \"CPQ RLDRAM single bit ECC error\" message reports a transient hardware issue. The cause is due to an ECC error detected in the MQCHIP CPQ memory. The error is self-correcting.";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
```
Comment (ConstrainedRandomVar) [inline juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set.rule]: Add count based rule here.

File: juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set.rule [hunk 15]
```diff
@@ -0,0 +1,201 @@
...
+                key-fields fpc_number;
+            }
+            pattern MQCHIP_EVENT009 {
+                event-id MQCHIP_EVENT009;
+                filter "fpc%{NUMBER:fpc_number}.*MQCHIP.*CPQ RLDRAM double bit ECC error.*bank.*addr";
+                description "The \"CPQ RLDRAM double bit ECC error\" message reports a transient hardware issue. The cause is due to a detected double bit error in ECC memory on the specified FPC.";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
```
Comment (ConstrainedRandomVar) [inline juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set.rule]: Add count based rule here.

File: juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set.rule [hunk 16]
```diff
@@ -0,0 +1,201 @@
...
+                key-fields fpc_number;
+            }
+            pattern MQCHIP_EVENT012 {
+                event-id MQCHIP_EVENT012;
+                filter "fpc%{NUMBER:fpc_number}.*MQCHIP.*CPQ Sram parity error.*errlog";
+                description "The \"CPQ Sram parity error\" message reports a transient hardware error.​ The cause is due to SRAM parity error in the CPQ block.";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
```
Comment (ConstrainedRandomVar) [inline juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set.rule]: Please add count based rule here.

File: juniper_official/Solutions/Pfe-wedge-detection/chassisd-pattern-set.rule (hunk 1 above)
Comment (d200201123) [reply juniper_official/Solutions/Pfe-wedge-detection/chassisd-pattern-set.rule]: @ConstrainedRandomVar , added trigger based on count

File: juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule (hunk 2 above)
Comment (d200201123) [reply juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule]: @ConstrainedRandomVar , added trigger based on count. Now count of syslog seen in last 3 days will be displayed.

File: juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule (hunk 3 above)
Comment (d200201123) [reply juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule]: @ConstrainedRandomVar , added trigger based on count

File: juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule (hunk 5 above)
Comment (d200201123) [reply juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule]: @ConstrainedRandomVar , added trigger based on count

File: juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule (hunk 6 above)
Comment (d200201123) [reply juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule]: @ConstrainedRandomVar , added trigger based on count

File: juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule (hunk 7 above)
Comment (d200201123) [reply juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule]: @ConstrainedRandomVar , added trigger based on count

File: juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule (hunk 8 above)
Comment (d200201123) [reply juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule]: @ConstrainedRandomVar , added trigger based on count

File: juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule (hunk 9 above)
Comment (d200201123) [reply juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule]: @ConstrainedRandomVar , added trigger based on count

File: juniper_official/Solutions/Pfe-wedge-detection/lueachip-pattern-set.rule (hunk 10 above)
Comment (d200201123) [reply juniper_official/Solutions/Pfe-wedge-detection/lueachip-pattern-set.rule]: @ConstrainedRandomVar , added trigger based on count

File: juniper_official/Solutions/Pfe-wedge-detection/lueachip-pattern-set.rule (hunk 11 above)
Comment (d200201123) [reply juniper_official/Solutions/Pfe-wedge-detection/lueachip-pattern-set.rule]: @ConstrainedRandomVar , added trigger based on count

File: juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set.rule (hunk 12 above)
Comment (d200201123) [reply juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set.rule]: @ConstrainedRandomVar , added trigger based on count

File: juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set.rule (hunk 13 above)
Comment (d200201123) [reply juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set.rule]: @ConstrainedRandomVar , added trigger based on count

File: juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set.rule (hunk 14 above)
Comment (d200201123) [reply juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set.rule]: @ConstrainedRandomVar , added trigger based on count

File: juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set.rule (hunk 15 above)
Comment (d200201123) [reply juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set.rule]: @ConstrainedRandomVar , added trigger based on count

**Comment by d200201123** [comment]
> 
> 
> Do we really care about AIS event rules? This requires a bundle of AIS op scripts loaded on the devices, and this communicates directly with JTAC? What additional value if any does HealthBot bring? Have dropped Sriram a note for his take. Based on that we can make a decision.
> 
> For the RMA recommendations, we should definitely have a count based rule.
> 
> For other rules which suggest an action followed by monitoring post that, I am not sure if there is a way that we can correlate it based on the corresponding syslog that the action would have generated, and then looking in the future beyond that..
> For example: " "Restart each fabric plane. If persists, then determine originating bad packets from stream mapping table and reset it. If logs recur after reset, contact JTAC.";"

@ConstrainedRandomVar , Removed the AIS rule

File: juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set.rule (hunk 16 above)
Comment (d200201123) [reply juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set.rule]: @ConstrainedRandomVar , added trigger based on count

**Comment by d200201123** [comment]
jenkins please retry a build

**Comment by d200201123** [comment]
> 
> 
> Hi Sarath,
> Can you please change the red-term and yellow-term time range of 7w to 3d?
> 
> Also, the way the rule is currently written, you can't accommodate two differing thresholds within the same pattern set. Can you please come up with a way to fix this - even if there are no Syslog RMAs that currently use it within the same pattern set?
> 
> Regards,
> 
> Harsha Lakshmikanth

Hi Harsha,

Reduced the hold time to 3 days and added Thresholds for chassisd patterns file and used them in rule triggers.

Please check.

Regards,
Sarath

File: juniper_official/Solutions/Pfe-wedge-detection/lueachip-pattern-set-syslog.ingest [hunk 17]
```diff
@@ -0,0 +1,84 @@
...
+                event-id LUEA_EVENT001;
+                filter "fpc%{NUMBER:fpc_number}.*lmem addr error";
+                description "The \"lmem addr error\" message is caused by a transient hardware or software defect. This error is caused by a transient hardware or software defect. ";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
+                    value "If single log, no action.  If numerous errors, reset the MPC and monitor for recurrence.";
+                }
+                constant reoccur {
```
Comment (ConstrainedRandomVar) [inline juniper_official/Solutions/Pfe-wedge-detection/lueachip-pattern-set-syslog.ingest]: Please add high and low thresholds here as well.

File: juniper_official/Solutions/Pfe-wedge-detection/lueachip-pattern-set-syslog.ingest [hunk 18]
```diff
@@ -0,0 +1,84 @@
...
+                event-id LUEA_EVENT005;
+                filter "fpc%{NUMBER:fpc_number}.*Uninitialized Read Error";
+                description "The \"Uninitialized Read Error\" message is caused by a transient hardware or software error.Access to unintialized memory with predefined patterns are software errors and the rest are hardware errors.";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
+                    value "No action if only a few occurrences.  If seen repeatedly, contact JTAC.";
+                }
+                constant reoccur {
```
Comment (ConstrainedRandomVar) [inline juniper_official/Solutions/Pfe-wedge-detection/lueachip-pattern-set-syslog.ingest]: And here.

File: juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set-syslog.ingest [hunk 19]
```diff
@@ -0,0 +1,249 @@
...
+                event-id MQCHIP_EVENT010;
+                filter "fpc%{NUMBER:fpc_number}.*MQCHIP.*FI Error-cell sent to reorder engine";
+                description "The \"FI Error-cell sent to reorder engine\" message reports a transient hardware issue. This is a transient hardware issue.  The MQCHIP Fabric output encountered a packet checksum error. ";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
+                    value "Restart each fabric plane.  If persists, then determine originating bad packets from stream mapping table and reset it.  If logs recur after reset, contact JTAC.";
+                }
+                constant reoccur {
```
Comment (ConstrainedRandomVar) [inline juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set-syslog.ingest]: Please add thresholds here.

File: juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set-syslog.ingest [hunk 20]
```diff
@@ -0,0 +1,249 @@
...
+            }
+            pattern MQCHIP_EVENT012 {
+                event-id MQCHIP_EVENT012;
+                filter "fpc%{NUMBER:fpc_number}.*MQCHIP.*CPQ Sram parity error.*errlog";
+                description "The \"CPQ Sram parity error\" message reports a transient hardware error.​ The cause is due to SRAM parity error in the CPQ block.";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
+                    value "Reset FPC during MW and monitor for recurrence.  If it recurs, then RMA.";
```
Comment (ConstrainedRandomVar) [inline juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set-syslog.ingest]: And here.

File: juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set-syslog.ingest [hunk 21]
```diff
@@ -0,0 +1,249 @@
...
+                event-id MQCHIP_EVENT013;
+                filter "fpc%{NUMBER:fpc_number}.*MQCHIP.*FI Enqueuing error.*type.*seq.*stream";
+                description "The \"FI Enqueuing error\" message reports a transient hardware error. This means that the fabric cells arrived too late in the MQ chip.";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
+                    value "Reset FPC during MW and monitor for recurrence.  If it recurs, then RMA.";
+                }
+                constant reoccur {
```
Comment (ConstrainedRandomVar) [inline juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set-syslog.ingest]: And here.

File: juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set-syslog.ingest [hunk 22]
```diff
@@ -0,0 +1,249 @@
...
+                event-id MQCHIP_EVENT014;
+                filter "fpc%{NUMBER:fpc_number}.*MQCHIP.*DDRIF WO Checksum Error";
+                description "The \"DDRIF WO Checksum Error\" message reports that there is a checksum error in the DDRIF memory.";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
+                    value "Reset FPC during MW and monitor for recurrence.  If it recurs, then RMA.";
+                }
+                constant reoccur {
```
Comment (ConstrainedRandomVar) [inline juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set-syslog.ingest]: And here..

File: juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set-syslog.ingest [hunk 23]
```diff
@@ -0,0 +1,249 @@
...
+                event-id MQCHIP_EVENT018b;
+                filter "fpc%{NUMBER:fpc_number}.*MQCHIP.*MALLOC Q Reference Count Memory parity error";
+                description "The \"MALLOC Pre-Q Reference Count underflow\" message reports a software error. The cause is due to an LU microcode error or misconfiguration. ";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
+                    value "If single log, no action.  If numerous errors, reset the MPC and monitor for recurrence.";
+                }
+                constant reoccur {
```
Comment (ConstrainedRandomVar) [inline juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set-syslog.ingest]: And here..

File: juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set-syslog.ingest [hunk 24]
```diff
@@ -0,0 +1,249 @@
...
+                event-id MQCHIP_EVENT007;
+                filter "fpc%{NUMBER:fpc_number}.*MQCHIP.*WI parity error detected when read from ibuf";
+                description "The \"WI parity error detected\" message reports a transient hardware issue. The cause is due to a parity error detected in the internal MQCHIP memory.";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
+                    value "If single log, clear alarm from CLI ('clear chassis fpc errors fpc-slot <fpc> all', only in 17.X+).  If multiple logs, reset FPC and monitor.";
+                }
+                constant reoccur {
```
Comment (ConstrainedRandomVar) [inline juniper_official/Solutions/Pfe-wedge-detection/mqchip-pattern-set-syslog.ingest]: And here..

**Pull request merged**